concent_api/core/message_handlers.py

```diff
@@ -954,6 +957,69 @@ def store_message(
     return stored_message
 
 
+def handle_send_subtask_results_verify(
+    request,
+    subtask_results_verify: message.concents.SubtaskResultsVerify
+):
+    subtask_results_rejected = subtask_results_verify.subtask_results_rejected
+    report_computed_task = subtask_results_rejected.report_computed_task
+    compute_task_def = report_computed_task.task_to_compute.compute_task_def
+    validate_id_value(
+        compute_task_def['subtask_id'],
+        'subtask_id'
+    )
+    validate_golem_message_task_to_compute(report_computed_task.task_to_compute)
+    validate_golem_message_subtask_results_rejected(subtask_results_rejected)
+
+    client_public_key = decode_client_public_key(request)
+    other_party_public_key = decode_other_party_public_key(request)
+    current_time = get_current_utc_timestamp()
+
+    if Subtask.objects.filter(
+        subtask_id=compute_task_def['subtask_id'],
+        state__in=[
+            Subtask.SubtaskState.VERIFICATION_FILE_TRANSFER.name,  # pylint: disable=no-member
+            Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name,     # pylint: disable=no-member
+        ]
+    ).exists():
+        return message.concents.ServiceRefused(
+            reason=message.concents.ServiceRefused.REASON.DuplicateRequest,
+        )
+    if current_time > subtask_results_rejected.timestamp + settings.ADDITIONAL_VERIFICATION_CALL_TIME:
+        return message.concents.ServiceRefused(
+            reason=message.concents.ServiceRefused.REASON.InvalidRequest,
+        )
+    if not base.is_requestor_account_status_positive(request):  # pylint: disable=no-value-for-parameter
+        return message.concents.ServiceRefused(
+            reason=message.concents.ServiceRefused.REASON.TooSmallRequestorDeposit,
+        )
+    if subtask_results_rejected.reason != SubtaskResultsRejected.REASON.VerificationNegative:
+        return message.concents.ServiceRefused(
+            reason=message.concents.ServiceRefused.REASON.InvalidRequest,
+        )
+
+    store_or_update_subtask(
+        task_id=compute_task_def['task_id'],
+        subtask_id=compute_task_def['subtask_id'],
+        provider_public_key=other_party_public_key,
+        requestor_public_key=client_public_key,
+        state=Subtask.SubtaskState.VERIFICATION_FILE_TRANSFER,
+        next_deadline=subtask_results_rejected.timestamp + settings.ADDITIONAL_VERIFICATION_CALL_TIME,
+        set_next_deadline=True,
+        task_to_compute=report_computed_task.task_to_compute,
+        subtask_results_rejected=subtask_results_rejected
+    )
+
+    send_blender_verification_request()
+
+    encoded_client_public_key = b64encode(decode_client_public_key(request))
+    ack_subtask_results_verify = message.concents.AckSubtaskResultsVerify(
+        subtask_results_verify=subtask_results_verify,
+        file_transfer_token=create_file_transfer_token(report_computed_task, encoded_client_public_key, "upload"),
```

Comment:
golem-messages has recently introduced an enum for `"download"` and `"upload"`. Might be a good idea to start using it.

https://github.com/golemfactory/golem-messages/blob/master/golem_messages/message/concents.py#L161-L164

Reply:
Since it is used in many places, there should be a sparate issue for it.